src/org/sosy_lab/java_smt/solvers/cvc5/CVC5AbstractProver.java

```diff
@@ -243,6 +249,22 @@ public Optional<List<BooleanFormula>> unsatCoreOverAssumptions(
     throw new UnsupportedOperationException();
   }
 
+  @Override
+  public ProofNode getProof() {
+
+    Proof[] proofs = solver.getProof();
+    if (proofs == null || proofs.length == 0) {
+      throw new IllegalStateException("No proof available");
+    }
+
+    // CVC5ProofProcessor pp = new CVC5ProofProcessor(creator);
+    try {
+      return CVC5ProofNode.fromCVC5Proof(proofs[0], creator);
+    } catch (CVC5ApiException e) {
+      throw new RuntimeException(e);
```

Comment:
Don't just throw `RuntimeException`s. This should be the `CVC5ApiException` wrapped in a `SolverException`, which is thrown in a checked way.